src/symbolize/mod.rs

```diff
@@ -36,10 +33,19 @@ use rustc_demangle::{try_demangle, Demangle};
 ///     });
 /// }
 /// ```
+#[cfg(feature = "std")]
+pub fn resolve<F: FnMut(&Symbol)>(addr: *mut c_void, mut cb: F) {
+    resolve_imp(addr as *mut _, &mut cb)
+}
+
+/// Without std this function now does not have synchronization guarentees.
+/// Please refer to std documentation for examples and explaination.
+#[cfg(not(feature = "std"))]
 pub fn resolve<F: FnMut(&Symbol)>(addr: *mut c_void, mut cb: F) {
```

Comment:
I think this forgot `unsafe`, but I'd personally prefer that we don't switch the signature of a function based on features enabled because that can mean it's a breaking change to activate a feature.

Could a new API be added, something like `resolve_unsynchronized`, which is `unsafe` and requires external synchronization? The safe `resolve` would then acquire a lock and call the raw API.

I believe this change would entail moving all locking from the inner modules to this function itself, but I think that's fine.